subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DefaultFileSystemSnapshotterTest.groovy

```diff
@@ -371,6 +375,45 @@ class DefaultFileSystemSnapshotterTest extends Specification {
 
         then:
         assertSingleFileSnapshot(snapshots)
+
+        when:
+        def tgzDir = tmpDir.createDir('tgzFile')
+        TestFile tgz = tempDir.file('emptyArchive.tgz');
+        tgzDir.tgzTo(tgz)
+        def localResource = new LocalResourceAdapter(TestFiles.fileRepository().localResource(tgz));
+        def emtpyBzipTree = TestFiles.fileOperations(tempDir, testFileProvider()).tarTree(localResource)
+        snapshots = snapshotter.snapshot(emtpyBzipTree)
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+        when:
+        def readableResource = new ReadableResource() {
+            @Override
+            InputStream read() throws MissingResourceException, ResourceException {
+                return file.newInputStream()
+            }
+
+            @Override
+            String getDisplayName() {
+                return "Some file based resource"
+            }
+
+            @Override
+            URI getURI() {
+                return file.toURI()
+            }
+
+            @Override
+            String getBaseName() {
+                return file.getName()
+            }
+        };
+        def recourceTarTree = TestFiles.fileOperations(tempDir, testFileProvider()).tarTree(readableResource)
+        snapshots = snapshotter.snapshot(recourceTarTree)
+
+        then:
+        snapshots.size() == 0
```

Comment:
There should be a snapshot - all the files in the tar as a directory snapshot. We don't want to ignore the `tarTree` input.